lib/expander/src/Events/CourseModuleCompleted.php

```diff
@@ -14,13 +9,11 @@ class CourseModuleCompleted extends  Event
     public function read(array $opts)
     {
 
-        global $DB;
-
-        $moduleid = $opts['contextinstanceid'];
 
-        $cm = $DB->get_record('course_modules',array('id' =>$moduleid));
-        $cmtype = $DB->get_record('modules',array('id' =>$cm->module));
+        $contextinstanceid = $opts['contextinstanceid'];
 
+        $cm = $this->repo->readModuleByContext($contextinstanceid);
```

Comment:
I like these names. They are close to moodle convention.   I don't know  others  more appropriate. Can you suggest other names  ?  :)